### PR TITLE
KAFKA-7817 ConsumerGroupCommand Regex Feature

### PR DESCRIPTION
**Pull request by rootex-**
### KAFKA-7817 ConsumerGroupCommand Regex Feature

**Description**: 
Add ability to select a subset of consumer groups using regex for operations: --describe, --delete and --reset-offsets
**Motivation**:
Suppose we have a subset of consumer groups with similar prefixes: group1, group2, ..., groupN and we want to apply operations (--describe/--delete/--reset) over [group1, group2, ..., group7] using CLI. 
It would be easier to declare a regex instead of listing all groups by name:
```bash
bin/kafka-consumer-groups.sh --bootstrap-server :9092 --describe      --regex group[1-7]
bin/kafka-consumer-groups.sh --bootstrap-server :9092 --delete        --regex group[1-7]
bin/kafka-consumer-groups.sh --bootstrap-server :9092 --reset-offsets --regex group[1-7] \
                             --topic topic --to-offset 50
```

**JIRA**: [Multiple Consumer Group Management with Regex](https://issues.apache.org/jira/browse/KAFKA-7817)
**Discussion 1**: [Re: Multiple Consumer Group Management](https://www.mail-archive.com/dev@kafka.apache.org/msg93781.html)
**Discussion 2**: [Re: ConsumerGroupCommand tool improvement?](https://www.mail-archive.com/dev@kafka.apache.org/msg90561.html)
**Tests:**: *Unit tests implemented*

### Committer Checklist (excluded from commit message)
- [x] Verify design and implementation 
- [x] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

File: core/src/main/scala/kafka/admin/ConsumerGroupCommand.scala [hunk 1]
```diff
@@ -58,7 +58,8 @@ object ConsumerGroupCommand extends Logging {
 
     try {
       if (opts.options.has(opts.listOpt))
-        consumerGroupService.listGroups().foreach(println(_))
+        if (opts.options.has(opts.regexOpt)) consumerGroupService.listGroupsByRegex().foreach(println(_))
```
Comment (abbccdda) [inline core/src/main/scala/kafka/admin/ConsumerGroupCommand.scala:61]: Do we also want to support regex for other commands as well?

**Review by abbccdda — COMMENTED** [review]
Thanks for the PR! One minor question I have is: do we have a place to print the groups being affected? It should be helpful for user to realize their change scope.

File: core/src/main/scala/kafka/admin/ConsumerGroupCommand.scala (hunk 1 above)
Comment (rootex-) [reply core/src/main/scala/kafka/admin/ConsumerGroupCommand.scala:61]: > Do we also want to support regex for other commands as well?

Not sure, which commands would we like to support?

**Comment by rootex-** [comment]
> Thanks for the PR! One minor question I have is: do we have a place to print the groups being affected? It should be helpful for user to realize their change scope.

Sure, standard output

**Comment by abbccdda** [comment]
@rootex- Do you still want to continue on this feature?

**Comment by rootex-** [comment]
@abbccdda Yes, sure

**Pull request closed without merging**